src/main/java/org/commonjava/indy/service/archive/util/HistoricalContentListReader.java

```diff
@@ -51,9 +53,16 @@ public HistoricalContentListReader( PreSeedConfig preSeedConfig )
             for ( HistoricalEntryDTO download : downloads )
             {
                 String path = download.getPath();
+                String packageType = download.getStoreKey().getPackageType();
+
+                if ( packageType.equals( NPM_PKG_KEY ) && !path.endsWith( ".tgz" ) )
```

Comment:
Do we can make sure that only ".tgz" is served for the non-metadata content in NPM? (I'm not sure if some packages will use .tar or .tar.gz to serve)

Reply:
@ligangty I think it should be yes so far, since I check non-tgz would be the package meta JSON content in Indy npm handler, based on https://github.com/Commonjava/indy/blob/master/addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java#L632

Reply:
So ok sounds good to me.